tscached/utils.py

```diff
@@ -105,6 +111,52 @@ def get_needed_absolute_time_range(time_range):
     return (start, end)
 
 
+def get_chunked_time_ranges(config, time_range):
+    """ Given a long kairos range, return N timestamp pairs so we can parallelize COLD calls (new->old).
+        This implements up to second precision.
+        :param config: dict, top-level tscached config.
+        :param time_range: dict, generated by populate_time_range containing kairos-formatted keys.
+        :return: list of 2-tuples of datetime.datetimes
+    """
+    chunk_length = config['chunking'].get('chunk_length', 1800)
+    num_chunks = config['chunking'].get('max_chunks', 6)
+    now = datetime.datetime.now()
+
+    start_time, end_time = get_needed_absolute_time_range(time_range, now)
+    if not end_time:
+        end_time = now
+    start_time = start_time.replace(microsecond=0)
+    end_time = end_time.replace(microsecond=0)
+
+    elapsed_secs = (end_time - start_time).total_seconds()
+    if elapsed_secs <= chunk_length:
+        return [(start_time, end_time)]
+    else:
+        # need to increase chunk length to fit into max chunks
+        if elapsed_secs > chunk_length * num_chunks:
+            chunk_length = int(elapsed_secs / num_chunks)
+        # need to downsize max chunks because we can use them of size chunk_length
+        else:
+            num_chunks = int(math.ceil(elapsed_secs / chunk_length))
+
+        chunks = []
+        length_td = datetime.timedelta(seconds=chunk_length)
+        # end_time is mutated below here
+        for i in xrange(num_chunks):
+            start = end_time - length_td
+
+            # Sanity check: make sure we limit the earliest chunk if it's partial
+            if start < start_time:
+                start = start_time
+
+            # Add an offset so the chunks don't overlap.
+            start += datetime.timedelta(seconds=1)
```

Comment:
Oh or is 1 sec the highest fidelity that kairos handles? in which case completely ignore my previous comment